pkg/handler/loki.go

```diff
@@ -31,7 +31,7 @@ func newLokiClient(cfg *loki.Config) httpclient.Caller {
 		}
 	}
 	// TODO: loki with auth
-	return httpclient.NewHTTPClient(cfg.Timeout, headers, cfg.SkipTLS)
+	return httpclient.NewHTTPClient(cfg.Timeout, headers, cfg.SkipTLS, cfg.UseMocks)
```

Comment:
rather than adding a param to `NewHTTPClient`, could we have the `if cfg.UseMocks` logic here?
It's just that I find it more logical to keep the lower level "http client" more agnostic to loki, whereas `loki.newLokiClient` is more legit to have this logic.

Reply:
Sounds legit ! 
Should I also move `HTTPClientJSONMock` to `handler` folder and rename it to something like `LokiClientMock` ?

Reply:
that would sound good, but I don't want to force you to refactor all :)